src/client/actions/ThreadListSync.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const Action = require('./Action');
+const Collection = require('../../util/Collection');
+const { Events } = require('../../util/Constants');
+
+class ThreadListSyncAction extends Action {
+  handle(data) {
+    const client = this.client;
+
+    const guild = client.guilds.cache.get(data.guild_id);
+    if (guild) {
```

Comment:
Might I suggest changing this line to:
```js
if (!guild) return {};
```
That way the next thirty or so lines don't need to be indented.